aws_lambda_powertools/logging/formatter.py

```diff
@@ -111,9 +111,15 @@ def __init__(
             Key-value to be included in log messages
 
         """
+
         self.json_deserializer = json_deserializer or json.loads
         self.json_default = json_default or str
-        self.json_serializer = json_serializer or partial(json.dumps, default=self.json_default, separators=(",", ":"))
+        self.json_indent = (
+            constants.PRETTY_INDENT if os.getenv("POWERTOOLS_DEV", "").lower() == "true" else constants.COMPACT_INDENT
```

Comment:
Tip: use [`strtobool`](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/aws_lambda_powertools/shared/functions.py#L9) as that accounts for variations like `"1"`, `"0"`, etc..

```suggestion
            constants.PRETTY_INDENT if strtobool(os.getenv("POWERTOOLS_DEV", "0")) else constants.COMPACT_INDENT
```

You'll need to import it like

```
from aws_lambda_powertools.shared.functions import strtobool
```